core/templates/dev/head/domain/summary/exploration-summary-backend-api.service.spec.ts

```diff
@@ -0,0 +1,188 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Unit tests for ExplorationSummaryBackendApiService.
+ */
+import { UpgradedServices } from 'services/UpgradedServices';
+
+require('domain/summary/exploration-summary-backend-api.service.ts');
+require('services/csrf-token.service.ts');
+
+describe('Exploration Summary Backend Api Service', function() {
+  var ExplorationSummaryBackendApiService = null;
+  var $httpBackend = null;
+  var $rootScope = null;
+  var CsrfService = null;
+  var AlertsService = null;
+
+  beforeEach(angular.mock.module('oppia'));
+  beforeEach(angular.mock.module('oppia', function($provide) {
+    var ugs = new UpgradedServices();
+    for (let [key, value] of Object.entries(ugs.getUpgradedServices())) {
+      $provide.value(key, value);
+    }
+  }));
+  beforeEach(angular.mock.inject(function($injector, $q) {
+    ExplorationSummaryBackendApiService = $injector.get(
+      'ExplorationSummaryBackendApiService');
+    $httpBackend = $injector.get('$httpBackend');
+    CsrfService = $injector.get('CsrfTokenService');
+    $rootScope = $injector.get('$rootScope');
+    AlertsService = $injector.get('AlertsService');
+
+    spyOn(CsrfService, 'getTokenAsync').and.callFake(function() {
+      var deferred = $q.defer();
+      deferred.resolve('sample-csrf-token');
+      return deferred.promise;
+    });
+  }));
+
+  afterEach(function() {
+    $httpBackend.verifyNoOutstandingExpectation();
+    $httpBackend.verifyNoOutstandingRequest();
+  });
+
+  it('should not load public exploration summaries from backend when' +
+    ' exploration id is not valid', function() {
+    var successHandler = jasmine.createSpy('success');
+    var failHandler = jasmine.createSpy('fail');
+    var explorationIds = ['#', null, '1'];
+    var alertSpy = spyOn(AlertsService, 'addWarning').and.callThrough();
+
+    ExplorationSummaryBackendApiService.loadPublicExplorationSummaries(
+      explorationIds).then(successHandler, failHandler);
+    $rootScope.$apply();
+
+    expect(alertSpy).toHaveBeenCalledWith(
+      'Please enter a valid exploration ID.');
+    expect(successHandler).not.toHaveBeenCalled();
+    expect(failHandler).toHaveBeenCalledWith([null, null, null]);
+  });
+
+  it('should load public exploration summaries from backend',
+    function() {
+      var successHandler = jasmine.createSpy('success');
+      var failHandler = jasmine.createSpy('fail');
+      var explorationIds = ['0', '1', '2'];
+      var sampleResults = [{
+        title: 'Title 1',
+        category: 'Category 1',
+        status: 'public',
+        language_code: 'en'
+      }];
+
+      var requestUrl = '/explorationsummarieshandler/data?' +
+        'stringified_exp_ids=0&' +
+        'stringified_exp_ids=1&' +
+        'stringified_exp_ids=2&' +
+        'include_private_explorations=false';
```

Comment:
This isn't the expected URL, right? I checked with the call locally, and this is what I am getting:
![image](https://user-images.githubusercontent.com/22347970/74460636-2f6ce680-4eb3-11ea-9b16-f42dbec9e306.png)
The exp_ids are comma separated...

Reply:
I just fixed the `requestUrl` to follow the same pattern as the call locally.